### PR TITLE
[Spark] DeltaAnalysis should not turn the target table of MERGE into v1 relation too early

### PR DESCRIPTION
**Pull request by cloud-fan**
## Description

In the Spark analyzer, it skips resolving the MERGE command if the target table is v2 and reports `ACCEPT_ANY_SCHEMA`. However, the `DeltaAnalysis` will turn Delta v2 relation to v1, and make Spark mistakenly resolve the MERGE commands targeting Delta tables, which can lead to issues as Delta supports more features like schema evolution.

We can't simply move the "turn Delta v2 relation to v1" code to the post-resolution phase, as we still need to do it to resolve metadata columns, which heavily rely on v1 relation. This PR does a surgical fix: if we find a MERGE command that can't be turned into its Delta variant yet (because the source query is not resolved), we mark its target table and do not turn it into v2 relation. After the MERGE command is fully resolved, we will turn it to the Delta variable, and all Delta relations inside it will become v1 relations
## How was this patch tested?

new test
## Does this PR introduce _any_ user-facing changes?

no

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @ryan-johnson-databricks @bart-samwel

**Review by ryan-johnson-databricks — COMMENTED** [review]
> However, the DeltaAnalysis will turn Delta v2 relation to v1, and makes Spark mistakenly resolve the Delta MERGE commands, which can lead to issues and Delta supports more features like schema evolution.

I had trouble understanding this, can you explain/reword?  If _think_ the situation is:
* spark analysis is designed to handle v2 tables that support schema evolution
* but DeltaAnalysis converts v2 to v1 in the first iteration of the "Resolution" batch
* the second iteration of the "Resolution" batch now sees a v1 relation and no longer allows schema evolution?

File: spark/src/main/scala/org/apache/spark/sql/delta/FallbackToV1Read.scala [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package main.scala.org.apache.spark.sql.delta
```
Comment (cloud-fan) [inline spark/src/main/scala/org/apache/spark/sql/delta/FallbackToV1Read.scala:18]:
```suggestion
package org.apache.spark.sql.delta
```

**Comment by cloud-fan** [comment]
@ryan-johnson-databricks DML resolution is tricky. Spark resolves DML commands, but Delta bypasses it as Delta has special features like schema evolution that are not in Spark yet. Delta has its own rules to resolve DML commands. In DS v2, we added an API to allow data sources to bypass Spark's DML resolution, and Delta uses this API. However, since Delta turns v2 relation to v1, this API doesn't work anymore and the Spark DML resolution is not bypassed.

**Review by ryan-johnson-databricks — APPROVED** [review]

**Pull request closed without merging**